File: euler-java/src/net/projecteuler/level001/Problem0007.java
package net.projecteuler.level001;

import net.projecteuler.utils.Utils;

/*
 * By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is
 * 13.
 *
 * What is the 10 001st prime number?
 */
public class Problem0007 {

  public static void main(String[] args) {

    int max = 10000000;

    int res = 0;

    boolean[] checked = new boolean[max];

    int count = 1;
    for (int i = 3; i < max; i = i + 2) {

      if (checked[i]) {
        continue;
      }

      boolean flag = true;
      for (int j = 2; j * j < i; j++) {
        if (i % j == 0) {
          flag = false;
          break;
        }
      }

      if (flag) {
        int m = i + i;
        while (m < max) {
          checked[m] = true;
          m = m + i;

        }
        count++;
        if (count == 10001) {
          System.out.println(i);
          res = i;
          break;
        }

      }

    }

    Utils.prime_factorization_print(Utils.doPrimeFactorization(res));

  }

}

File: euler-scala/src/net/projecteuler/level001/Problem0007.scala
package net.projecteuler.level001

import net.projecteuler.utils.{Prime, Stopwatch}

/*
By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see that the 6th prime is 13.

What is the 10 001st prime number?
 */
object Problem0007 {

  def main(args: Array[String]): Unit = {

    Stopwatch.elapsed(brute_force)

  }

  def brute_force() = {
    var n = 0
    var i = 0
    while(n < 10001) {

      i+= 1
      if(Prime.isPrime(i)) {
        n += 1
      }
    }

    println(i)
  }

}
